aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -166,259 +224,333 @@ module aptos_framework::fungible_asset {
+        ref.metadata
+    }
 
-    /// Return `allowed_ungated_transfer`.
-    public fun ungated_transfer_allowed(wallet: &Object<FungibleAssetWallet>): bool acquires FungibleAssetWallet {
-        borrow_fungible_asset(wallet).allow_ungated_transfer
+    /// Get the underlying metadata object from `BurnRef`.
+    public fun burn_ref_metadata(ref: &BurnRef): Object<FungibleAssetMetadata> {
+        ref.metadata
+    }
+
+    /// Transfer `amount` of fungible metadata of `metadata` to `receiver`.
+    /// Note: it does not move the underlying object.
+    public entry fun transfer<T: key>(
+        sender: &signer,
+        metadata: Object<T>,
+        amount: u64,
+        recipient: address,
+    ) acquires FungibleAssetWallet, FungibleAssetWalletEvents {
+        let fa = withdraw(sender, metadata, amount);
+        deposit(recipient, fa);
+    }
+
+    /// Withdraw `amount` of fungible asset from an account's wallet.
+    public fun withdraw<T: key>(
+        owner: &signer,
+        metadata: Object<T>,
+        amount: u64,
+    ): FungibleAsset acquires FungibleAssetWallet, FungibleAssetWalletEvents {
+        let owner_addr = signer::address_of(owner);
+        assert!(wallet_exists(owner_addr, metadata), error::invalid_argument(EINSUFFICIENT_BALANCE));
+        assert!(
+            ungated_transfer_allowed(owner_addr, metadata),
+            error::invalid_argument(EUNGATED_TRANSFER_IS_NOT_ALLOWED),
+        );
+        let wallet = create_wallet_address_from_account(owner_addr, metadata);
+        withdraw_internal(wallet, amount)
+    }
+
+    /// Deposit fungible assets into an account.
+    public fun deposit(owner: address, fa: FungibleAsset) acquires FungibleAssetWallet, FungibleAssetWalletEvents {
+        assert!(
+            !wallet_exists(owner, fa.metadata) ||
+                ungated_transfer_allowed(owner, fa.metadata),
+            error::invalid_argument(EUNGATED_TRANSFER_IS_NOT_ALLOWED),
+        );
+        deposit_internal(owner, fa);
     }
 
-    /// Mint the `amount` of fungible asset.
+    /// Mint the specified `amount` of fungible asset.
     public fun mint(
         ref: &MintRef,
         amount: u64,
-    ): FungibleAsset acquires FungibleAssetMetadata {
+    ): FungibleAsset acquires FungibleAssetMetadata, FungibleAssetMetadataEvents {
         assert!(amount != 0, error::invalid_argument(EAMOUNT_CANNOT_BE_ZERO));
-        let metadata = verify(&ref.metadata);
-        increase_supply(&ref.metadata, amount);
+        let metadata = ref.metadata;
+        increase_supply(&metadata, amount);
+
+        let events = borrow_global_mut<FungibleAssetMetadataEvents>(object::object_address(&metadata));
+        event::emit_event(&mut events.mint_events, MintEvent { amount });
+
         FungibleAsset {
             metadata,
             amount
         }
     }
 
+    /// Mint the specified `amount` of fungible asset to a destination address.
+    public fun mint_to(
```

Comment:
seems not required with one line code.

Reply:
It's just a style preference, but usually when I do multi-line functions, I usually put each argument on a separate line for more readability (with the exceptions of tests where readability matters a bit less)